Tasks/Common/nuget-task-common/NuGetToolRunner.ts

```diff
@@ -91,29 +91,53 @@ export function createNuGetToolRunner(nuGetExePath: string, settings: NuGetEnvir
     return runner;
 }
 
-export function locateTool(tool: string, userPath?: string, optional?: boolean) {
-    tl.debug("looking for tool " + tool);
-    if (userPath) {
-        tl.debug("using user-supplied path " + userPath)
-        tl.checkPath(userPath, 'nuget');
-        return userPath;
-    }
+export interface LocateOptions {
+    optional?: boolean;
+    userPath?: string;
+    fallbackToSystemPath?: boolean;
+}
 
+export function locateTool(tool: string | string[], opts?: LocateOptions) {
```

Comment:
The interface is a bit confusing. `fallbackToSystemPath`, `optional`, and `tool[1..N-1]` are never considered when `userPath` is specified.

Comments would be helpful for readers.